### ISSUE TITLE
codeAt in HumanStandardTokenFactory.sol does not work

### ISSUE THREAD

**Issue by skmgoldin** [ConsenSys/Tokens#57]
This is the function:
```
    function codeAt(address _addr) internal returns (bytes o_code) {
      assembly {
          // retrieve the size of the code, this needs assembly
          let size := extcodesize(_addr)
          // allocate output byte array - this could also be done without assembly
          // by using o_code = new bytes(size)
          o_code := mload(0x40)
          // new "memory end" including padding
          mstore(0x40, add(o_code, and(add(add(size, 0x20), 0x1f), not(0x1f))))
          // store length in memory
          mstore(o_code, size)
          // actually retrieve the code, this needs assembly
          extcodecopy(_addr, add(o_code, 0x20), 0, size)
      }
    }
```
Its test fails. ~~I think I have an idea as to why.~~

~~Lets enumerate the magic numbers in the line `mstore(0x40, add(o_code, and(add(add(size, 0x20), 0x1f), not(0x1f))))`~~

~~`0x40` is the location of the free memory pointer.~~
~~`0x20` is 32 bytes, which we want to leave free at the beginning of `o_code` so we can store the buffer size in it (`mstore(o_code, size)`).~~
~~`0x1f` _appears_ to be a number with which we determine the amount of memory to reserve. Though its exact nature is mysterious to me.~~

~~Imagine `size` is 0. `0x20 + 0x1f` is `0x3f`. In decimal, the 32 byte size field plus the 31 byte magic number is 63 bytes. The negation of `0x1f` (`not(0x1f)`) is `0xe0` (224). `and(0x3f, 0xe0)` is `0x20`, or 32 bytes, which is what we would expect if we had an empty buffer: a size element storing zero followed by no data.~~

~~If the code is one byte in size and we do this math again we get 64 bytes, which also makes sense. A 32 byte size element followed by a 32 byte data element. If the code size is 33 bytes we'll get 96, which also makes sense: a 32 byte size element followed by two 32 byte data elements.~~

~~This all works until you get to code sizes greater than 192 bytes (which would correctly give you a 224 byte buffer). If you have a code size of 193 bytes, `and(0x100, 0xe0)` will give you 0.~~

```
0000 0001 0000 0000
0000 0000 1110 0000
```
~~And the answers will continue to be funky for all code sizes greater than 192 bytes.~~

~~So I think we need to understand where the magic number `0x1f` comes from, and how we can make it not a magic number on the basis of the size of the code being examined.~~

**Comment by skmgoldin** [comment]
This whole theory is wrong. See: https://consensys.slack.com/archives/C03DVPNTT/p1500487032466679

**Comment by skmgoldin** [comment]
This is still a bug though! For whatever it's worth, if somebody else's eyes see something, calling the getter for humanStandardByteCode returns undefined. Writing a custom getter to get the length also returns undefined (in JavaScript world).

**Comment by simondlr** [comment]
That's odd. It definitely worked when I tested it originally. Will retest.

**Comment by skmgoldin** [comment]
Addressed in #58.
